modules/mauth-authenticator/src/main/java/com/mdsol/mauth/RequestAuthenticator.java

```diff
@@ -128,18 +128,23 @@ private boolean validateSignatureV1 (MAuthRequest mAuthRequest, PublicKey client
     byte[] decryptedSignature = MAuthSignatureHelper.decryptSignature(clientPublicKey, mAuthRequest.getRequestSignature());
 
     // Recreate the plain text signature, based on the incoming request parameters, and hash it.
-    String unencryptedRequestString =
-        MAuthSignatureHelper.generateUnencryptedSignature(
-            mAuthRequest.getAppUUID(), mAuthRequest.getHttpMethod(), mAuthRequest.getResourcePath(),
-            new String(mAuthRequest.getMessagePayload(), StandardCharsets.UTF_8),
-            String.valueOf(mAuthRequest.getRequestTime())
-        );
-    byte[] messageDigest_bytes = MAuthSignatureHelper
-        .getHexEncodedDigestedString(unencryptedRequestString).getBytes(StandardCharsets.UTF_8);
+    try {
+      byte[] messageDigest_bytes = MAuthSignatureHelper.generateUnencryptedSignature(
+          mAuthRequest.getAppUUID(), mAuthRequest.getHttpMethod(), mAuthRequest.getResourcePath(),
+          mAuthRequest.getMessagePayload(),
+          String.valueOf(mAuthRequest.getRequestTime())
+      );
+      messageDigest_bytes = MAuthSignatureHelper.getHexEncodedDigestedString(messageDigest_bytes).getBytes(StandardCharsets.UTF_8);
+
+      // Compare the decrypted signature and the recreated signature hashes.
+      // If both match, the request was signed by the requesting application and is valid.
+      return Arrays.equals(messageDigest_bytes, decryptedSignature);
+    } catch (Exception ex) {
+      final String message = "MAuth request validation failed because of " + ex.getMessage();
+      logger.error(message);
+      throw new MAuthValidationException(message);
```

Comment:
probably adding `MAuthValidationException(String message, Throwable cause)` constructor and use that instead of passing just the message of the trowable would be better

Reply:
Updated in https://github.com/mdsol/mauth-jvm-clients/pull/96/commits/7de735546c1694e3da46b41c7a6ff4d0268c7005